### PR TITLE
minor: removes checkstyle specific class in ReverseListIterator

### PR DESCRIPTION
**Pull request by rnveach**
`ReverseListIterator` is basically a utility class. It shouldn't have anything specific to checkstyle similar to `ReverseDescendantIterator` and the others...

### CODE REVIEW

File: src/main/java/com/puppycrawl/tools/checkstyle/xpath/iterators/ReverseListIterator.java [hunk 1]
```diff
@@ -22,7 +22,6 @@
 import java.util.ArrayList;
 import java.util.List;
 
-import com.puppycrawl.tools.checkstyle.xpath.AbstractNode;
```
Comment (romani) [inline src/main/java/com/puppycrawl/tools/checkstyle/xpath/iterators/ReverseListIterator.java:25]: Can we support not usage by import control Check ?

**Review by romani — CHANGES_REQUESTED** [review]
Items

**Review by nick-mancuso — APPROVED** [review]

File: src/main/java/com/puppycrawl/tools/checkstyle/xpath/iterators/ReverseListIterator.java (hunk 1 above)
Comment (rnveach) [reply src/main/java/com/puppycrawl/tools/checkstyle/xpath/iterators/ReverseListIterator.java:25]: Done.

**Review by romani — APPROVED** [review]
Ok to merge

**Pull request merged**